docs/preflight/v1beta3-guide.md

```diff
@@ -0,0 +1,487 @@
+      - warn:
+          when: 'min(memoryCapacity) < {{ .Values.nodeResources.memory.recommendedGi }}Gi'
+          message: Recommended {{ .Values.nodeResources.memory.recommendedGi }} GiB per node
+      - pass:
+          message: All nodes meet recommended memory
+
+  # Per-node ephemeral storage (Gi)
+  nodeResources:
+    checkName: Per-node ephemeral storage
+    outcomes:
+      - fail:
+          when: 'min(ephemeralStorageCapacity) < {{ .Values.nodeResources.ephemeral.minGi }}Gi'
+          message: All nodes must have at least {{ .Values.nodeResources.ephemeral.minGi }} GiB
+      - warn:
+          when: 'min(ephemeralStorageCapacity) < {{ .Values.nodeResources.ephemeral.recommendedGi }}Gi'
+          message: Recommended {{ .Values.nodeResources.ephemeral.recommendedGi }} GiB per node
+      - pass:
+          message: All nodes meet recommended ephemeral storage
+  ```
+
+- **deploymentStatus**: verify workload deployment status
+  ```yaml
+  deploymentStatus:
+    checkName: Deployment ready
+    namespace: '{{ .Values.workloads.deployments.namespace }}'
+    name: '{{ .Values.workloads.deployments.name }}'
+    outcomes:
+      - fail:
+          when: absent
+          message: Deployment not found
+      - fail:
+          when: '< {{ .Values.workloads.deployments.minReady }}'
+          message: Deployment has insufficient ready replicas
+      - pass:
+          when: '>= {{ .Values.workloads.deployments.minReady }}'
+          message: Deployment has sufficient ready replicas
+  ```
+
+- **postgres/mysql/redis**: database connectivity (requires collectors)
+  ```yaml
+  # Collector section
+  - postgres:
+      collectorName: '{{ .Values.databases.postgres.collectorName }}'
+      uri: '{{ .Values.databases.postgres.uri }}'
+
+  # Analyzer section
+  postgres:
+    checkName: Postgres checks
+    collectorName: '{{ .Values.databases.postgres.collectorName }}'
+    outcomes:
+      - fail:
+          message: Postgres checks failed
+      - pass:
+          message: Postgres checks passed
+  ```
+
+- **textAnalyze/yamlCompare/jsonCompare**: analyze collected data
+  ```yaml
+  textAnalyze:
+    checkName: Text analyze
+    collectorName: 'cluster-resources'
+    fileName: '{{ .Values.textAnalyze.fileName }}'
+    regex: '{{ .Values.textAnalyze.regex }}'
+    outcomes:
+      - fail:
+          message: Pattern matched in files
+      - pass:
+          message: Pattern not found
+  ```
+
+
+### Design conventions for maintainability
+
+- **Guard every optional analyzer** with a values toggle, so consumers can enable only what they need.
+- **Always include collectors section** when analyzers require them (databases, http, registryImages, etc.).
+- **Use `checkName`** to provide a stable, user-facing label for each check.
+- **Prefer `fail` for unmet hard requirements**, `warn` for soft requirements, and `pass` with a direct, affirmative message.
```

Comment:
@paigecalvert i think some of these which are conventions independent of `v1beta2` should probably come up into a more general introduction to Preflights doc. I think it just goes back to my other comment on what the correct information architecture is of how someone would progress through these docs.